app/src/authFetch.ts

```diff
@@ -40,9 +46,9 @@ export async function authFetch(
   throw new Error("An unexpected error occurred while fetching data");
 }
 
-let refreshPromise: Promise<Response> | null = null;
+let refreshPromise: Promise<Response | null> | null = null;
```

Comment:
rather than null can we make the return something more declarative to be readable.